src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/readthreading/ReadThreadingAssembler.java

```diff
@@ -78,7 +78,7 @@ public ReadThreadingAssembler(final int maxAllowedPathsForReadThreadingAssembler
                                   final int maxUnprunedVariants, final boolean useLinkedDebruijnGraphs) {
         Utils.validateArg( maxAllowedPathsForReadThreadingAssembler >= 1, "numBestHaplotypesPerGraph should be >= 1 but got " + maxAllowedPathsForReadThreadingAssembler);
         this.kmerSizes = new ArrayList<>(kmerSizes);
-        kmerSizes.sort(Integer::compareTo);
+        this.kmerSizes.sort(Integer::compareTo);
```

Comment:
this is such a hard to notice mistake.  I wonder if intellij can catch this for us...  
You could change it to 

`this.kmerSizes = kmerSizes.stream().sort().collect(Collectors.toList())` if we wanted to avoid the sort call.

Reply:
You know i think its actually cleaner and a little more future proof to do this in one line so i'll take that.